src/highlighter.rs

```diff
@@ -0,0 +1,80 @@
+use {
+    nu_ansi_term::{Color, Style},
+    std::borrow::Cow,
+};
+
+pub static DEFAULT_BUFFER_MATCH_COLOR: Color = Color::Green;
+pub static DEFAULT_BUFFER_NEUTRAL_COLOR: Color = Color::White;
+pub static DEFAULT_BUFFER_NOTMATCH_COLOR: Color = Color::Red;
+
+pub trait Highlighter {
+    fn highlight<'l>(&self, line: &'l str) -> Cow<'l, str>;
+}
+
+pub struct DefaultHighlighter {
+    external_commands: Vec<String>,
+    match_color: Color,
+    notmatch_color: Color,
+    neutral_color: Color,
```

Comment:
I'm still of the opinion that these should be `NuStyle`s as [defined here](https://github.com/jonathandturner/reedline/blob/54215568952db347f8bdffdcf830d0fa59fa4022/src/syntax_highlighting_fileio.rs#L10-L14) instead of `Color`. Perhaps in a later PR we can update this and read these colors from a config file using `syntax_highlighting_fileio.rs`.